Automattic/wp-calypso
# Start User: Add Continue as user to the sign in flow

Instead of showing a notice that is not very helpful when the user clicks the browser back button on the domain screen.

Let's show the user a Continue as User component instead.

#### Changes proposed in this Pull Request

When the "user has completed the registration step" has been detected. We redirect the user to the registration form (same url) but with the ?user_completed=1 query parameter in the URL that helps us know if a user really want to be on this page. Usually, we skip the user registration step if they are already logged in. 


#### Testing instructions
See 

https://user-images.githubusercontent.com/115071/142338815-70a426a7-2ce9-4646-8e10-95ac44d16af6.mp4

* Visit 
/start 
Create a new user. 
Once you are on the domain Screen. Use the browser back button. 

Notice that you are logged in as a newly created user. 
* Click continue - you should end up on the domain screen again
* Click the browser back button. 
* Click Login in as someone else 
- notice that you end up on the /start/user screen. 
- You are now logged out. 

Other things to test
visit /log-in and /log-in/new notice that it looks as expected.

Related to https://github.com/Automattic/wp-calypso/issues/57287
